src/Features/OnboardingTasks.php

```diff
@@ -134,29 +138,40 @@ public static function check_task_completion( $task ) {
 			case 'products':
 				$products = wp_count_posts( 'product' );
 				return (int) $products->publish > 0 || (int) $products->draft > 0;
-
 			case 'homepage':
-				// @todo This should be run client-side in a Gutenberg hook and add a notice
-				// to return to the task list if complete.
 				$homepage_id = get_option( 'woocommerce_onboarding_homepage_post_id', false );
-
 				if ( ! $homepage_id ) {
 					return false;
 				}
-
 				$post      = get_post( $homepage_id );
 				$completed = $post && 'publish' === $post->post_status;
-				if ( $completed ) {
-					update_option( 'show_on_front', 'page' );
-					update_option( 'page_on_front', $homepage_id );
-				}
-
 				return $completed;
 		}
-
 		return false;
 	}
 
+	/**
+	 * Updates the product published message with a continue setup link, if the products task is currently active.
+	 */
+	function update_product_success_message( $messages ) {
+		if ( ! $this->check_active_task_completion() ) {
+			return $messages;
+		}
+		/* translators: 1: onboarding task list url */
+		$messages['product'][6] = sprintf( __( 'You created your first product! <a href="%s">Continue Setup</a>.', 'woocommerce-admin' ), esc_url( wc_admin_url() ) );
+		return $messages;
+	}
+
+	/**
+	 * Hooks into the post page to display a different success notice and sets the active page as the site's home page if visted from onboarding.
+	 */
+	function add_onboarding_homepage_notice_admin_script( $hook ) {
+		global $post;
+		if ( $hook == 'post.php' && 'page' === $post->post_type && isset( $_GET[ 'wc-onboarding' ] ) && 'homepage' === $_GET[ 'wc-onboarding' ] ) { // WPCS: csrf ok.
```

Comment:
Nitpick refactor: maybe we should create a method `get_active_task()` to check for the `wc-onboarding` get param in case the logic for this changes again.

Reply:
I've actually backed out my change the the parameter here. There's no reason not to use the same query parameter as `self::ACTIVE_TASK_TRANSIENT`.